client/tests/utils.py

```diff
@@ -153,7 +153,7 @@ def create_client_job(recipe_dir, name="TestJob", sleep=1):
     # as well as BUILD_ROOT replacement
     utils.create_recipe_environment(name="GLOBAL_NAME", value="BUILD_ROOT/global", recipe=recipe)
     count = 0
-    for s in ["step0", "step1", "step2"]:
+    for s in ["step{}".format(i) for i in range(n_steps)]:
```

Comment:
I learned about format strings a while back, perfect for this
```suggestion
    for s in [f"step{i}" for i in range(n_steps)]:
```